chainerx_cc/chainerx/backward_builder.cc

```diff
@@ -154,12 +154,32 @@ RetainedInputToken BackwardBuilder::RetainInput(size_t input_index) {
     return {internal::GetArrayBody(gsl::at(inputs_, input_index))->GetParams(), input_index};
 }
 
+std::vector<RetainedInputToken> BackwardBuilder::RetainInput(std::vector<size_t> indices) {
+    std::vector<RetainedInputToken> token;
+    for (size_t i : indices) {
+        CHAINERX_ASSERT(i < inputs_.size());
+        input_retention_record_.Record(i);
+        token.push_back({internal::GetArrayBody(gsl::at(inputs_, i))->GetParams(), i});
```

Comment:
`push_back` might be fine but we use `emplace_back` consistently in other places.
Let's use `emplace_back`.